underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -361,6 +361,11 @@ public void cleanup() {
     mManager.abortMultipartUploads(mBucketName, cleanBefore);
   }
 
+  @Override
+  public void close() {
+    mExecutor.shutdown();
```

Comment:
just want to make sure this `mExecutor` is not passed in or shared right?

Reply:
It is used by the multipart upload. This is a safe shutdown so it will wait for any live tasks that have not finished, but any upload objects that have been created but not yet started will fail.

Actually it looks like the UFS class isn't actually be closed in all places where it is created either, so there might be other things to fix as well.